src/frontend/apps/impress/src/features/docs/doc-management/api/useRemoveFavoriteDoc.tsx

```diff
@@ -0,0 +1,59 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { APIError, errorCauses, fetchAPI } from '@/api';
+import { Doc, KEY_LIST_DOC } from '@/features/docs';
+
+export type RemoveFavoriteDocParams = Pick<Doc, 'id'>;
+
+export const removeFavoriteDoc = async ({
+  id,
+}: RemoveFavoriteDocParams): Promise<void> => {
+  const response = await fetchAPI(`documents/${id}/favorite/`, {
+    method: 'DELETE',
+  });
+
+  if (!response.ok) {
+    throw new APIError(
+      'Failed to remove the doc as favorite',
+      await errorCauses(response),
+    );
+  }
+  const data: unknown = await response.json();
+
+  return data as Promise<void>;
+};
+
+interface RemoveFavoriteDocProps {
+  onSuccess?: () => void;
+  listInvalideQueries?: string[];
+}
+
+export function useRemoveFavoriteDoc({
+  onSuccess,
+  listInvalideQueries,
+}: RemoveFavoriteDocProps = {}) {
+  const queryClient = useQueryClient();
+  return useMutation<void, APIError, RemoveFavoriteDocParams>({
+    mutationFn: removeFavoriteDoc,
+    onSettled: () => {
+      void queryClient.invalidateQueries({
+        queryKey: ['favorites'],
+      });
+      void queryClient.invalidateQueries({
+        queryKey: [KEY_LIST_DOC],
+      });
+    },
+    onSuccess: () => {
+      void queryClient.invalidateQueries({
+        queryKey: ['favorites'],
```

Comment:
Let's keep it homogeneous, same for the others:
```suggestion
        queryKey: [KEY_LIST_DOC_FAVORITE],
```